src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -182,24 +203,37 @@ public class ReleaseService : IReleaseService
                 .OnSuccessDo(async () => await _releaseFileService.DeleteAll(releaseId))
                 .OnSuccessVoid(async release =>
                 {
+                    release.SoftDeleted = true;
+                    _context.Update(release);
+                    
                     var roles = await _context
                         .UserReleaseRoles
                         .Where(r => r.ReleaseId == releaseId)
                         .ToListAsync();
-
+                    roles.ForEach(r => r.SoftDeleted = true);
+                    _context.UpdateRange(roles);
+                    
                     var invites = await _context
                         .UserReleaseInvites
                         .Where(r => r.ReleaseId == releaseId)
                         .ToListAsync();
-
-                    release.SoftDeleted = true;
-                    roles.ForEach(r => r.SoftDeleted = true);
                     invites.ForEach(r => r.SoftDeleted = true);
-
-                    _context.Update(release);
-                    _context.UpdateRange(roles);
                     _context.UpdateRange(invites);
 
+                    var methodologiesScheduledWithRelease = 
+                        GetMethodologiesScheduledWithRelease(releaseId);
+                    
+                    methodologiesScheduledWithRelease.ForEach(m =>
+                    {
+                        m.PublishingStrategy = Immediately;
```

Comment:
If you're like me, you'd see all this logic in ReleaseService and think that ReleaseService now knows a bit too much about Methodology Status setting.

I was going to add another Repository that could be shared between the ReleaseService and MethodologyService to avoid this shared logic, but then I didn't want to add yet another dependency to ReleaseService and also it would be a repo with just one method in it - "UpdateStatusToDraft".

I could've internalised this logic into a method on MethodologyVersion called "SetToDraft()" as an alternative which then both services could reuse, but then we do precious little model-driven code in this project so it would be a little out of the norm.

Interested to know what you might think!